src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/RocketChatAsyncHelper.java

```diff
@@ -0,0 +1,98 @@
+import de.caritas.cob.userservice.mailservice.generated.web.model.ErrorMailDTO;
+import de.caritas.cob.userservice.mailservice.generated.web.model.TemplateDataDTO;
+import java.util.List;
+import java.util.function.Consumer;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RocketChatAsyncHelper {
+
+  private final @NonNull RocketChatFacade rocketChatFacade;
+  private final @NonNull SessionRepository sessionRepository;
+  private final @NonNull IdentityClient identityClient;
+  private final @NonNull ConsultingTypeManager consultingTypeManager;
+  private final @NonNull ConsultantRepository consultantRepository;
+  private final @NonNull MailService mailService;
+  @Value("${app.base.url}")
+  private String applicationBaseUrl;
+
+  @Async
+  @Transactional
+  public void addConsultantToSessions(Consultant consultant, AgencyDTO agency,
+      Consumer<String> logMethod, Long tenantId) {
+    try {
+      TenantContext.setCurrentTenant(tenantId);
+      List<Session> relevantSessions = collectRelevantSessionsToAddConsultant(agency);
+      RocketChatAddToGroupOperationService
+          .getInstance(this.rocketChatFacade, identityClient, logMethod,
+              consultingTypeManager)
+          .onSessions(relevantSessions)
+          .withConsultant(consultant)
+          .addToGroupsOrRollbackOnFailure();
+      updateConsultantStatus(consultant, ConsultantStatus.CREATED);
+    } catch (Exception e) {
+      updateConsultantStatus(consultant, ConsultantStatus.ERROR);
+      sendErrorEmail(consultant, e);
+      log.error("Error happened during rocket chat session assignments", e);
+    }
+    TenantContext.clear();
+  }
+
+  private void updateConsultantStatus(Consultant consultant, ConsultantStatus error) {
+    consultant.setStatus(error);
+    consultantRepository.save(consultant);
+  }
+
+  private void sendErrorEmail(Consultant consultant, Exception exception) {
+    ErrorMailDTO errorMailDTO = new ErrorMailDTO()
+        .template(TEMPLATE_FREE_TEXT)
+        .templateData(asList(
+            new TemplateDataDTO().key("subject").value("RocketChat sessions assignment error"),
+            new TemplateDataDTO().key("url").value(this.applicationBaseUrl),
+            new TemplateDataDTO().key("text").value(getEmailText(consultant, exception))));
+    this.mailService.sendErrorEmailNotification(errorMailDTO);
+  }
+
+  private String getEmailText(Consultant consultant, Exception exception) {
+    return "Error happened during rocket chat session assignments for consultant " + consultant
+        .getUsername() + "Error message: " + exception.getMessage();
```

Comment:
missing space before "Error message:" it will be joined without space.  Besides better to use some string formatter to format the message. I would also suggest to put message template into properties